operator/src/main/java/org/keycloak/operator/controllers/config/DeploymentConfig.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.operator.controllers.config;
+
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import org.keycloak.operator.controllers.KeycloakDeployment;
+import org.keycloak.operator.crds.v2alpha1.deployment.KeycloakStatusBuilder;
+
+/**
+ * An interface to represent a configuration of the particular deployment
+ *
+ * @author <a href="mailto:mabartos@redhat.com">Martin Bartos</a>
+ */
+public interface DeploymentConfig {
```

Comment:
At first when see this, I really liked this approach, but the I started wondering whether we really need this in the end, what will be the use cases for it. Currently, there's hostname and TLS. The logic for hostname will be simplified as we drop `INSECURE-DISABLE` as part of #14395 to align it with the dist. So there won't be more special logic needed, rest of it is handled by `KeycloakIngress`. In general, we should limit custom logic around config as much as possible and rather implement the changes at the dist side. Most of the fields will be just pass-through without the need for any special logic. And the few selected cases like TLS maybe can stay inside the Deployment?